CRM/Report/Form.php

```diff
@@ -4893,4 +4893,26 @@ public function joinGroupTempTable($baseTable, $field, $tableAlias) {
     }
   }
 
+  /**
+   * Get all labels for fields that are used in a group concat.
+   *
+   * @param string $options
+   *   comma separated option values.
+   * @param array $group
+   *   array of option value/label pairs.
+   *
+   * return string
+   */
+  public function getGroupLabels($options, $group) {
```

Comment:
@eileenmcnaughton The reason why I refrained from using the individual getLabel function to retrieve the labels was because I needed an array containing all the labels to perform a non-duplicate check.

For e.g. 
Contributions for a whole year can have multiple transactions, many of which use the same card type. The record would then look like this: 

![duplicate](https://cloud.githubusercontent.com/assets/1990869/23987115/d9a149d2-0a4d-11e7-9dd6-f14c9b004bce.png)

I'm not sure if using getLabel function in the foreach, then removing the duplicate labels is the way to go here. Could you please instruct otherwise?